ir/src/constraint_builder/expression.rs

```diff
@@ -139,27 +130,28 @@ impl ConstraintBuilder {
     // --- IDENTIFIER EXPRESSIONS -----------------------------------------------------------------
 
     /// Adds a trace column, periodic column, random value, named constant or a variable to the
-    /// graph and returns the [ExprDetails] of the inserted expression.
+    /// graph and returns the [NodeIndex] of the inserted expression.
     ///
     /// # Errors
     /// Returns an error if the identifier is not present in the symbol table or is not a supported
     /// type.
-    fn insert_symbol_access(
+    fn insert_binding_access(
         &mut self,
-        name: &str,
-        access_type: AccessType,
+        binding_access: BindingAccess,
     ) -> Result<NodeIndex, SemanticError> {
```

Comment:
Following my prior comments, I'd probably keep the old name and change the signature to look something like:
```Rust
fn insert_symbol_access(
    &mut self,
    symbol_access: SymbolAccess
) -> Result<NodeIndex, SemanticError>
```